#14 - Shopping cart Testing Implementation

waspp01 opened the pull request:
- Implemented testing for ShoppingCartServiceImpl and ShoppingCartController
- Fixed annotation issue in UpdateCartItemRequestDto class
- Added @Accessors(chain = true) where needed
- Added needed sql files for tests

waspp01 committed:
#14 - Shopping cart Testing Implementation
- Implemented testing for ShoppingCartServiceImpl and ShoppingCartController
- Fixed annotation issue in UpdateCartItemRequestDto class
- Added @Accessors(chain = true) where needed
- Added needed sql files for tests

sokimaaa commented on src/main/java/com/example/onlinebookstore/dto/shoppingcart/ShoppingCartDto.java:
```diff
@@ -3,8 +3,10 @@
 import com.example.onlinebookstore.dto.cartitem.CartItemDto;
 import java.util.Set;
 import lombok.Data;
+import lombok.experimental.Accessors;
```
never use lombok.experimental.* cause it is unstable

sokimaaa commented on src/test/java/com/example/onlinebookstore/controller/ShoppingCartControllerTest.java:
```diff
@@ -0,0 +1,208 @@
+                .andReturn();
+
+        //then
+        ShoppingCartDto actual = objectMapper.readValue(
+                result.getResponse().getContentAsString(),
+                ShoppingCartDto.class);
```
dont use object mapper, spring boot test has `jsonpath` to simplify\optimize your tests
```suggestion;
```

sokimaaa commented on src/test/java/com/example/onlinebookstore/util/cartitem/TestCartItemSupplier.java:
```diff
@@ -0,0 +1,9 @@
+
+import com.example.onlinebookstore.dto.cartitem.CreateCartItemRequestDto;
+
+public class TestCartItemSupplier {
+    public static CreateCartItemRequestDto getTestCreateCartItemRequestDto() {
+        return new CreateCartItemRequestDto().setBookId(2L).setQuantity(2);
+    }
+}
```
useless
```suggestion
```

sokimaaa commented on src/test/java/com/example/onlinebookstore/util/shoppingcart/TestShoppingCartSupplier.java:
```diff
@@ -0,0 +1,38 @@
+import java.util.List;
+import java.util.Set;
+
+public class TestShoppingCartSupplier {
```
useless

waspp01 committed:
#14 - Shopping cart Testing Implementation
- Implemented jsonPath usage instead of ObjectMapper
- Deleted redundant TestCartItemSupplier class
- Added needed dependency in pom.xml
